alacritty_terminal/src/term/mod.rs

```diff
@@ -1802,35 +1793,46 @@ impl<T: EventListener> Handler for Term<T> {
         self.color_modified[index] = false;
     }
 
-    /// Set the clipboard.
+    /// Store data into clipboard.
     #[inline]
-    fn set_clipboard(&mut self, clipboard: u8, base64: &[u8]) {
+    fn clipboard_store(&mut self, clipboard: u8, base64: &[u8]) {
         let clipboard_type = match clipboard {
             b'c' => ClipboardType::Clipboard,
             b'p' | b's' => ClipboardType::Selection,
             _ => return,
         };
 
-        if let Ok(bytes) = base64::decode(base64) {
-            if let Ok(text) = str::from_utf8(&bytes) {
-                self.clipboard.store(clipboard_type, text);
-            }
-        }
+        let bytes = match base64::decode(base64) {
+            Ok(bytes) => bytes,
+            Err(_) => return,
+        };
+
+        let text = match String::from_utf8(bytes) {
+            Ok(text) => text,
+            Err(_) => return,
+        };
```

Comment:
Why didn't you handle this using `and_then` as suggested?

Reply:
`and_then` doesn't work because of different error types, since it requires the original error type, however error type changes, so it's a map of both error type and `Ok` value. So you can't use `and_then`.

Reply:
Ah right, now I remember why I didn't change it in my pager implementation. I thought originally it was because of borrowing/lifetime constraints.

So you should either change it back or do something like the following:

```
if let Some(text) = base64::decode(base64).ok().and_then(|bytes| String::from_utf8(&bytes).ok())
```

Probably easier to just nest the two conditionals. Another option would be to use `String::from_utf8_lossy`, since base64 is always guaranteed to be utf8. Alternatively `from_utf8_unchecked` would also work just fine here, but I'm not sure if we should make this unsafe when there really is no need to.